packages/react-devtools-extensions/src/injectGlobalHook.js

```diff
@@ -67,24 +65,23 @@ window.__REACT_DEVTOOLS_GLOBAL_HOOK__.nativeWeakMap = WeakMap;
 window.__REACT_DEVTOOLS_GLOBAL_HOOK__.nativeSet = Set;
 `;
 
-// If we have just reloaded to profile, we need to inject the renderer interface before the app loads.
-if (sessionStorageGetItem(SESSION_STORAGE_RELOAD_AND_PROFILE_KEY) === 'true') {
-  const rendererURL = chrome.runtime.getURL('build/renderer.js');
-  let rendererCode;
+// We need to inject the renderer interface before the app loads so we can patch the
+// console before initial render.
```

Comment:
It's _probably_ okay but I'm curious, did you do any performance testing to see how much this slows down a page load? (Is it true that we'll always do this now, even when the DevTools panel isn't open?)

cc @DotProto The sync XHR pattern we talked about is spreading 😃 Is it still okay?

Reply:
Sorry, my memory is failing me. I remember talking about sync XHR in abstract, but I'm super hazy on the details. Based on the surrounding comments, it looks like the basic use case is to inject a content script in a host page in order to monkey patch console.log before the React app on the page initializes. Is that correct?

If so, yeah, I don't think there's a better way to accomplish this in Chrome. Hopefully we can more directly address this use case by expanding the dynamic content script capabilities in `chrome.scripting`. I'm not up on the details, but I think Firefox's existing [`browser.contentScripts`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/contentScripts) API would also serve this use case.

Reply:
Oh hi @dotproto!

Yeah, in some cases (e.g. reload-and-profile) DevTools needs to _synchronously_ inject a script that it normally injects asynchronously (in order to avoid missing initial mount times).

This PR/feature will require us to do that more often. Turns out it's a little more costly than we'd hoped though so we're looking into ways of pairing down what we sync inject for the common case. (Reload and profile won't change.)